### PR TITLE
Tests: Adjust views and form tests to use Django's client

### PR DESCRIPTION
**Pull request by Yoavsc129**
I only changed the tests for tips_views based on https://github.com/redhat-beyond/Xpense/issues/90, https://github.com/redhat-beyond/Xpense/issues/98

### CODE REVIEW

File: tips/tests/tips_views_tests.py [hunk 1]
```diff
@@ -80,12 +66,11 @@ def test_bad_form(self, generate_bad_form):
...
 @pytest.mark.django_db
 class TestTipViews:
-    def test_tips_board_function_200(self, generate_get_request, generate_tip):
-        try:
-            render = board(generate_get_request)
-            assert render.status_code == 200, "status code is not 200"
-        except Exception:
-            assert False, "Error in board function"
+    def test_tips_board_function_200(self):
+        c = Client()
```
Comment (ShellyGolden) [inline tips/tests/tips_views_tests.py:70]: I think that client instead of c is better

File: tips/tests/tips_views_tests.py [hunk 2]
```diff
@@ -101,7 +86,9 @@ def test_tips_views_templates(self):
         except TemplateDoesNotExist:
             assert False, f"Template {EDIT_TIP_PAGE_ROUTE} does not exist"
 
-    def test_delete_tip_page(self, generate_get_request, generate_tip):
+    def test_delete_tip_page(self, generate_tip):
         generate_tip.save()
-        delete_tip(generate_get_request, generate_tip.id)
+        assert generate_tip in Tip.objects.all()
+        c = Client()
```
Comment (ShellyGolden) [inline tips/tests/tips_views_tests.py:92]: also here

File: tips/tests/tips_views_tests.py (hunk 1 above)
Comment (ronyyosef) [reply tips/tests/tips_views_tests.py:70]: Also, you can use the fixture of the client
It is a little bit cleaner

**Comment by ronyyosef** [comment]
What about all the other tests in our project? are you going to change them too?

File: tips/tests/tips_views_tests.py [hunk 3]
```diff
@@ -101,7 +86,9 @@ def test_tips_views_templates(self):
         except TemplateDoesNotExist:
             assert False, f"Template {EDIT_TIP_PAGE_ROUTE} does not exist"
 
-    def test_delete_tip_page(self, generate_get_request, generate_tip):
+    def test_delete_tip_page(self, generate_tip):
```
Comment (LiorKGOW) [inline tips/tests/tips_views_tests.py:89]: The name of this test confused me a bit
Does this test delete a tip from the page?
Or does it delete the page of the tips?
Please change it to a more meaningful name

**Review by LiorKGOW — CHANGES_REQUESTED** [review]
GJ Yoav please address the comments 👍 
Left you a comment nothing more to add from all the things that already came up

**Comment by Yoavsc129** [comment]
> What about all the other tests in our project? are you going to change them too?

I will get there, it will more or less look the same, removing the requestfactory

File: tips/tests/tips_views_tests.py (hunk 2 above)
Comment (Yoavsc129) [reply tips/tests/tips_views_tests.py]: done

File: tips/tests/tips_views_tests.py (hunk 1 above)
Comment (Yoavsc129) [reply tips/tests/tips_views_tests.py]: done

File: tips/tests/tips_views_tests.py (hunk 3 above)
Comment (Yoavsc129) [reply tips/tests/tips_views_tests.py]: done

**Comment by shlior7** [comment]
Hi Yoav great work!
two things,
maybe we should add a test for editing a Tip as well (title, content, category etc separately or together)
Im gonna move these test to the new tests folder and remove the duplicated fixtures as well just wanted to let you know

**Comment by Yoavsc129** [comment]
> Since I just merged #65 there are changes in the files and caused some conflicts. Please rebase and resolve conflicts. Make sure that you are using the right files after rebasing.If there were changes in fixtures that were moved make sure you are copy those to the right place.

**Review by svatares — APPROVED** [review]
Sorry it took so long. Had a lot of things going on at work

**Pull request merged**